### PR TITLE
Display more clearly when registration is full

### PR DESCRIPTION
**Pull request by DanielEgdal**
Per this comment ( https://github.com/thewca/worldcubeassociation.org/issues/3810#issuecomment-1200439737) I tried coming up with a solution

### CODE REVIEW

File: WcaOnRails/app/models/competition.rb [hunk 1]
```diff
@@ -395,6 +395,11 @@ def warnings_for(user)
       if self.results.any? && !self.results_posted?
         warnings[:results] = I18n.t('competitions.messages.results_not_posted')
       end
+
+      if self.registration_full? && registration_opened?
```
Comment (gregorbg) [inline WcaOnRails/app/models/competition.rb:399]: Nit: `self.registration_opened?`

**Review by gregorbg — COMMENTED** [review]
Thanks for your contribution! Do you have a screenshot of how your new warning message looks like?

**Comment by DanielEgdal** [comment]
I don't have the website running locally, but the warning will look the same as other warnings displayed on the front page of competitions. Like this one: 
image

**Comment by gregorbg** [comment]
Okay, I'll accept that as a monkey-patch screenshot :D

**Review by gregorbg — APPROVED** [review]

**Comment by DanielEgdal** [comment]
Hmm, I just checked this on the website and saw there is an error. I don't quite know how to fix it to be displayed correctly, like it is on the registration page
image
registration page:
image

**Comment by gregorbg** [comment]
Ah, my bad, I forgot to check the actual translation value during review.
You need what's called an "interpolation", or in other words to tell the translation engine which values to insert into the `%{gaps}` in pre-translated strings. To do so, you have to invoke the `I18n.t()` method like this:

```ruby
I18n.t('registrations.registration_full', competitor_limit: self.competitor_limit)
```

Where the key name `competitor_limit` (the first occurence before the colon) stems from the fact that the original string in the translations file is

```yaml
"The competitor limit for this competition is %{competitor_limit} and it has already been reached. If you register, your registration will be placed on a waiting list. It may not be approved unless a spot frees up."
```

**Comment by KerrieJarman** [comment]
The %(competitor_limit) message is still showing - can it be either fixed or rolled back ?

**Comment by gregorbg** [comment]
Hotfixed, might take a minute or two to deploy.

**Comment by DanielEgdal** [comment]
Thanks :)

**Pull request merged**